net/softethervpn/Makefile

```diff
@@ -0,0 +1,113 @@
+with OpenVPN, L2TP, IPsec, EtherIP, L2TPv3, Cisco VPN Routers and MS-SSTP VPN Clients. SoftEther VPN is the world's only VPN software which supports SSL-VPN, 
+OpenVPN, L2TP, EtherIP, L2TPv3 and IPsec as a single VPN software. SoftEther VPN is not only an alternative VPN server to existing VPN products (OpenVPN, 
+IPsec and MS-SSTP), but has also original strong SSL-VPN protocol to penetrate any kinds of firewalls. Ultra-optimized SSL-VPN Protocol of SoftEther VPN
+has very fast throughput, low latency and firewall resistance.
+endef
+
+
+define Package/softethervpn/conffiles
+/usr/bin/softethervpn/vpn_server.config
+/usr/bin/softethervpn/vpn_client.config
+/usr/bin/softethervpn/vpn_bridge.config
+/usr/bin/softethervpn/lang.config
+endef
+
+
+ifeq ($(CONFIG_OPENSSL_WITH_SSL3),y)
+	SSL3_SUPP=-DSSL3_SUPPORT
+endif
+
+
+define Build/Configure
+	if [ -v CONFIG_64BIT ]; then cp $(PKG_BUILD_DIR)/src/makefiles/linux_64bit.mak $(PKG_BUILD_DIR)/Makefile; else cp $(PKG_BUILD_DIR)/src/makefiles/linux_32bit.mak $(PKG_BUILD_DIR)/Makefile; fi;
+endef
+
+define Build/Compile
+	# Softether compiles first a tool called hamcorebuilder which is essentially an archiver. Then the compilation calls this archiver to pack all 
+	# resource files (*.html, *.config, .txt, readme, languages) it needs to run in an archive file called hamcore.se2. The three executables are then built 
+	# and they will have a runtime dependency on this resource archive. By default OpenWRT build process cross compiles all targets in the makefile
+	# generating therefore an error when the archive is generated since the executable is crosscompiled to run on the router and instead it is run
+	# by the build host. The workaround is to compile the archiver with the host environment, generate the hamcore.se2 archive and cross compile everything
+	# again without regenerating the archive.
+
+	# we clean the last cross compiled version
+	make -C $(PKG_BUILD_DIR) clean
+
+	# we compile the hamcorebuilder utility and generate hamcore archive using host environment
+	make -C $(PKG_BUILD_DIR) src/bin/BuiltHamcoreFiles/unix/hamcore.se2
+
+	# we clean everything, but we save the built hamcore archive and we update its timestamp to avoid rebuilding the archive with cross compiled hamcorebuilder
+	mv $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2 $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2.1
+	make -C $(PKG_BUILD_DIR) clean
+		mv $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2.1 $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2
+	touch -d "`date -d 1day`" $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2
+
+	ls -la $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix
+
+	# we cross compile everything from the scratch
+	$(MAKE) \
+		$(TARGET_CONFIGURE_OPTS) \
+		CCFLAGS="$(SSL3_SUPP) -I$(STAGING_DIR)/usr/include $(ICONV_CFLAGS)" \
+		LDFLAGS="-L$(STAGING_DIR)/usr/lib $(ICONV_LDFLAGS) -liconv" \
```

Comment:
Do not hardcode `-liconv` see http://lists.en.qi-hardware.com/pipermail/discussion/2011-February/007092.html